src/features/broker-protection/actions/expectation.js

```diff
@@ -7,24 +7,135 @@ import { ErrorResponse, SuccessResponse } from '../types.js'
  * @return {import('../types.js').ActionResponse}
  */
 export function expectation (action, root = document) {
-    const expectations = action.expectations
+    const results = expectMany(action.expectations, root)
 
-    const allExpectationsMatch = expectations.every(expectation => {
-        if (expectation.type === 'text') {
-            // get the target element text
-            const elem = getElement(root, expectation.selector)
-            return Boolean(elem?.textContent?.includes(expectation.expect))
-        } else if (expectation.type === 'url') {
-            const url = window.location.href
-            return url.includes(expectation.expect)
-        }
+    const errors = results.filter(x => x.result === false).map(x => {
+        if ('error' in x) return x.error
+        return 'unknown error'
+    })
+
+    if (errors.length > 0) {
+        return new ErrorResponse({ actionID: action.id, message: errors.join(', ') })
+    }
+
+    return new SuccessResponse({ actionID: action.id, actionType: action.actionType, response: null })
+}
 
-        return false
+/**
+ * Return a true/false result for every expectation
+ *
+ * @param {import("../types").Expectation[]} expectations
+ * @param {Document | HTMLElement} root
+ * @return {import("../types").BooleanResult[]}
+ */
+export function expectMany (expectations, root) {
+    return expectations.map(expectation => {
+        switch (expectation.type) {
```

Comment:
Worth a test to validate non-happy path cases here (invalid or missing type) to ensure we're getting back what we expect?